src/app/features/activity-list/activity-list.tsx

```diff
@@ -42,14 +66,21 @@ export function ActivityList() {
   const isBitcoinEnabled = useConfigBitcoinEnabled();
 
   const isInitialLoading =
-    isInitialLoadingBitcoinTransactions ||
+    isInitialLoadingNsBitcoinTransactions ||
+    isInitialLoadingTrBitcoinTransactions ||
     isInitialLoadingStacksTransactions ||
     isInitialLoadingStacksPendingTransactions;
 
-  const transactionListBitcoinTxs = useMemo(
-    () => convertBitcoinTxsToListType(bitcoinTransactions),
-    [bitcoinTransactions]
-  );
+  const transactionListBitcoinTxs = useMemo(() => {
+    const filteredTxs: BitcoinTx[] = [];
+    [...nsBitcoinTransactions, ...trBitcoinTransactions].forEach(tx => {
+      if (!filteredTxs.find(t => t.txid === tx.txid)) {
+        filteredTxs.push(tx);
+      }
```

Comment:
can probs figure out what it's doing but would rather read a descriptive method name

Reply:
actually, seems better to use lodash uniqby here